vllm/v1/worker/gpu_input_batch.py

```diff
@@ -195,6 +195,10 @@ def __init__(
         self.logit_bias: List[Optional[Dict[int,
                                             float]]] = [None] * max_num_reqs
 
+        self.has_allowed_token_ids: List[bool] = [False] * max_num_reqs
+        self.allowed_token_ids_mask: torch.Tensor = torch.zeros(
+            max_num_reqs, self.vocab_size, dtype=torch.bool, device=device)
```

Comment:
This could actually get quite large (e.g. 128 MB), we may want to consider a different (pre)allocation strategy.

Also, consistent with the other tensors, we should keep this in pinned CPU memory and only copy to GPU in an async manner when the batch makeup changes (and has_allowed_token_ids is True)